packages/store/src/schematics/src/state/index.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Rule,
+  SchematicContext,
+  Tree,
+  apply,
+  chain,
+  mergeWith,
+  template,
+  url,
+} from '@angular-devkit/schematics';
+import { classify, dasherize } from '@angular-devkit/core/src/utils/strings';
+
+const stringUtils = { classify, dasherize };
+
+export default function (options: any): Rule {
+  return chain([
+    (_tree: Tree, context: SchematicContext) => {
+      context.logger.info('My Full Schematic: ' + JSON.stringify(options));
+    },
+
+    mergeWith(apply(url('./files'), [
+      template({
+        ...stringUtils,
+        name: options.name,
+      }),
+    ])),
+  ]);
+}
+
+/*
+TODO
```

Comment:
See this - https://github.com/angular/material2/blob/0a501586a5a72feb924dd3b2b3bae890454bd1cc/src/lib/schematics/utils/ast.ts#L26-L53